shared/team-building/search-result/you-result.tsx

```diff
@@ -12,14 +12,20 @@ const YouResult = React.memo((props: ResultProps) => {
     dispatch(TeamBuildingGen.createCancelTeamBuilding({namespace: 'chat2'}))
     dispatch(Chat2Gen.createPreviewConversation({participants: [props.username], reason: 'search'}))
   }
-  return (
-    <CommonResult
-      {...props}
-      onAdd={onSelfChat}
-      rowStyle={styles.rowContainer}
-      bottomRow={<Kb.Text type="BodySmall">Write secure notes to yourself</Kb.Text>}
-    />
-  )
+
+  let bottomRow: React.ReactNode = null
+
+  if (props.namespace === 'teams') {
+    bottomRow = (
+      <Kb.Text type="BodySmall">
+        {props.isPreExistingTeamMember ? 'Already in team' : 'Add yourself to the team'}
+      </Kb.Text>
+    )
+  } else {
+    bottomRow = <Kb.Text type="BodySmall">Write secure notes to yourself</Kb.Text>
+  }
+
+  return <CommonResult {...props} onAdd={onSelfChat} rowStyle={styles.rowContainer} bottomRow={bottomRow} />
```

Comment:
I think `onAdd` should have the regular behavior outside of the chat context, rn it always takes you to your self chat

Reply:
fixed. also fixed clicking on add in wallets which calls cancel twice and closes the modal!